### PR TITLE
wire in the otel converter and create a migration guide

### LINKED ISSUE #6419: Add support for converting OpenTelemetry Collector components
grafana/agent#6379 introduced a framework for converting OpenTelemetry Collector components, with initial support for converting `otlpreceiver` and `otlpexporter`. There are many more components we need to support conversions for. The following lists track OpenTelemetry Collector components used in both static mode or exclusively used in Flow mode. (Flow mode also uses the components from static mode).

> **NOTE**: Some components may not make sense to convert, but are left in this list for completion. Such components can be ignored and checked off.
>
> Components which do not have a direct Flow component, or are used as an implementation detail in static mode, may be ignored and revisited later during work on converting the tracing subsystem for static mode.

```[tasklist]
### General
- [ ] https://github.com/grafana/agent/issues/6697
- [ ] https://github.com/grafana/agent/issues/6698
```

```[tasklist]
### Components used in static mode
- [ ] https://github.com/grafana/agent/issues/6420
- [ ] https://github.com/grafana/agent/issues/6421
- [ ] https://github.com/grafana/agent/issues/6422
- [ ] https://github.com/grafana/agent/issues/6423
- [ ] https://github.com/grafana/agent/issues/6424
- [ ] https://github.com/grafana/agent/issues/6425
- [ ] https://github.com/grafana/agent/issues/6428
- [ ] https://github.com/grafana/agent/issues/6429
- [ ] https://github.com/grafana/agent/issues/6437
- [ ] https://github.com/grafana/agent/issues/6432
- [ ] https://github.com/grafana/agent/issues/6433
- [x] Support converting `otlpreceiver`
- [x] Support converting `otlpexporter`
- [ ] https://github.com/grafana/agent/issues/6430
- [ ] https://github.com/grafana/agent/issues/6431
- [ ] https://github.com/grafana/agent/issues/6434
- [ ] https://github.com/grafana/agent/issues/6426
- [ ] https://github.com/grafana/agent/issues/6427
- [ ] https://github.com/grafana/agent/issues/6435
- [ ] https://github.com/grafana/agent/issues/6436
- [ ] https://github.com/grafana/agent/issues/6438
- [ ] https://github.com/grafana/agent/issues/6718
```

```[tasklist]
### Components used exclusively in flow mode
- [ ] https://github.com/grafana/agent/issues/6439
- [ ] https://github.com/grafana/agent/issues/6440
- [ ] https://github.com/grafana/agent/issues/6441
- [ ] https://github.com/grafana/agent/issues/6442
- [ ] https://github.com/grafana/agent/issues/6443
- [ ] https://github.com/grafana/agent/issues/6444
- [ ] https://github.com/grafana/agent/issues/6445
- [ ] https://github.com/grafana/agent/issues/6451
- [ ] https://github.com/grafana/agent/issues/6452
- [ ] https://github.com/grafana/agent/issues/6453
- [ ] https://github.com/grafana/agent/issues/6454
- [ ] https://github.com/grafana/agent/issues/6455
- [ ] https://github.com/grafana/agent/issues/6456
- [ ] https://github.com/grafana/agent/issues/6458
- [ ] https://github.com/grafana/agent/issues/6459
- [ ] https://github.com/grafana/agent/issues/6448
```

### LINKED ISSUE #6697: Add otelcol as a supported converter format to the CLI and update docs

### LINKED ISSUE #6698: Create a migration guide for otelcol to flow

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### Which issue(s) this PR fixes

Closes #6697 
Closes #6698 
Closes #6419

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [ ] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (rfratto) [inline docs/sources/flow/tasks/migrate/from-otelcol.md:247]: Should we make it clearer that we don't support conversions for components that we do not embed directly?

**Review by rfratto — APPROVED** [review]
LGTM but really needs an approval from Clayton before merging

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (erikbaranowski) [reply docs/sources/flow/tasks/migrate/from-otelcol.md]: I added some words about this.

File: docs/sources/flow/reference/cli/convert.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/convert.md:127]:
```suggestion
You can use the `--source-format=otelcol` to convert the source configuration from an [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/configuration/) to a {{< param "PRODUCT_NAME" >}} configuration.
```

File: docs/sources/flow/reference/cli/convert.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/convert.md:139]:
```suggestion
Many OpenTelemetry Collector components are supported.
Review the `otelcol.*` component information in the [Component Reference][] for more information about `otelcol` components that you can convert.
If a source configuration has unsupported features, you will receive [errors] when you convert it to a flow configuration.
The converter raises warnings for configuration options that may require your attention.
```

File: docs/sources/flow/reference/cli/convert.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/convert.md:141]:
```suggestion
Refer to [Migrate from OpenTelemetry Collector to {{< param "PRODUCT_NAME" >}}][migrate] for a detailed migration guide.

[Component Reference]: ../../components/
[migrate]: ../../../tasks/migrate/from-otelcol/

```

We should not use relrefs in any documentation now. This shortcode has been deprecated. We can use relative URIs for links within the same doc tree which should work in both the OSS and Cloud docs.

@erikbaranowski We need to check the paths in a local build once these changes are applied. The URI sometimes has a slightly different path depth than relrefs.

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (clayton-cornell) [inline docs/sources/flow/tasks/migrate/from-otelcol.md:7]:
```suggestion
canonical: https://grafana.com/docs/agent/latest/flow/tasks/migrate/from-otelcol/
```
New topic doesn't need aliases defined. These are only required if we move or rename a previously published topic.

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (clayton-cornell) [inline docs/sources/flow/tasks/migrate/from-otelcol.md:128]:
```suggestion
> using an OpenTelemetry Collector configuration.
```

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (clayton-cornell) [inline docs/sources/flow/tasks/migrate/from-otelcol.md:66]:
```suggestion
1. If the `convert` command can't convert an OpenTelemetry Collector configuration, diagnostic information is sent to `stderr`.\
```

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (clayton-cornell) [inline docs/sources/flow/tasks/migrate/from-otelcol.md:149]:
```suggestion
This example demonstrates converting an OpenTelemetry Collector configuration file to a {{< param "PRODUCT_NAME" >}} configuration file.
```

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (clayton-cornell) [inline docs/sources/flow/tasks/migrate/from-otelcol.md]: Same change suggestion here as in previous file.

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (clayton-cornell) [inline docs/sources/flow/tasks/migrate/from-otelcol.md:259]: These doc references can stay as-is.. or be converted to relative URIs.

For example: 
```
[Components]: "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/concepts/components.md"
[Components]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/send-data/agent/flow/concepts/components.md"
```

would instead be:
```
[Components]: ../../../concepts/components/
```
(we would have to verify the directory depth)

**Review by clayton-cornell — APPROVED** [review]
Approving - as discussed to replace the relrefs and we are good to go. :-)

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (tpaschalis) [inline docs/sources/flow/tasks/migrate/from-otelcol.md:57]: Should we include an example line with the run command and OUTPUT_CONFIG_PATH?

**Review by tpaschalis — APPROVED** [review]
Great work man, you're a steamroller!

File: docs/sources/flow/tasks/migrate/from-otelcol.md
Comment (erikbaranowski) [reply docs/sources/flow/tasks/migrate/from-otelcol.md:57]: We thought about that on the other converter tasks and chose to link to the Run docs since the command can vary quite a bit depending on the environment.

**Pull request merged**
